src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -869,11 +878,27 @@ else if (itemChildName.equals("value")) {
         return tags;
     }
 
-    protected QuestionDef parseControl(IFormElement parent, Element e, int controlType) {
-        return parseControl (parent, e, controlType, null);
+    private QuestionDef parseControl(IFormElement parent, Element e, int controlType) {
```

Comment:
Just making sure you’re aware that the return value from this method is never used. I would make the return type void, but I understand if you have a different preference.

Reply:
Oh? I'm seeing it used in `parseUpload`.

Reply:
The second parseControl is called from parseUpload, no?

![screen shot 2018-03-19 at 6 37 35 pm](https://user-images.githubusercontent.com/55099/37630961-a3ad48b6-2ba4-11e8-8ad1-6c84f2b5d99f.png)